src/main/webapp/app/editor/instructions/editor-instructions.component.ts

```diff
@@ -0,0 +1,434 @@
+    setupMarkDown() {
+        this.markDown = new Remarkable();
+        // TODO: check if bind required
+        this.markDown.inline.ruler.before('text', 'testsStatus', this.remarkableTestsStatusParser.bind(this), {});
+        this.markDown.block.ruler.before('paragraph', 'plantUml', this.remarkablePlantUmlParser.bind(this), {});
+        this.markDown.renderer.rules['testsStatus'] = this.remarkableTestsStatusRenderer.bind(this);
+        this.markDown.renderer.rules['plantUml'] = this.remarkablePlantUmlRenderer.bind(this);
+    }
+
+    /**
+     * @function renderReadme
+     * @desc Prepares and starts the rendering process of the README.md file
+     */
+    renderReadme() {
+        this.isLoading = true;
+        // Reset steps array
+        this.steps = [];
+        // Render README.md file via Remarkable
+        this.readMeFileRendered = this.markDown.render(this.readMeFileContent);
+        this.isLoading = false;
+
+        // Detach test status click listeners if already initialized
+        if (this.listenerRemoveFunctions && this.listenerRemoveFunctions.length) {
+            this.removeTestStatusClickListeners();
+        }
+
+        // Since our rendered markdown file gets inserted into the DOM after compile time, we need to register click events for test cases manually
+        const testStatusDOMElements = this.elRef.nativeElement.querySelectorAll('.test-status');
+        testStatusDOMElements.forEach( element => {
+            const listenerRemoveFunction = this.renderer.listen(element, 'click', event => {
+                // Extract the data attribute for tests and open the details popup with it
+                const tests = event.target.parentElement.getAttribute('data-tests');
+                this.showDetailsForTests(this.latestResult, tests);
+            });
+            this.listenerRemoveFunctions.push(listenerRemoveFunction);
+        });
+
+        if (!this.loadedDetails) {
+            this.loadResultsDetails();
+        }
+    }
+
+    /**
+     * @function removeTestStatusClickListeners
+     * @desc Detaches all click listeners for test status links
+     */
+    removeTestStatusClickListeners() {
+        for (const listenerRemoveFunction of this.listenerRemoveFunctions) {
+            // Call each removal function to detach the click listener from DOM
+            listenerRemoveFunction();
+        }
+        // Set function array to empty
+        this.listenerRemoveFunctions = [];
+    }
+
+    /**
+     * @function remarkablePlantUmlParser
+     * @desc Parser rule for Remarkable custom token PlantUml
+     * @param state
+     * @param startLine
+     * @param endLine
+     * @param silent
+     */
+    remarkablePlantUmlParser = function(state, startLine, endLine, silent) {
+        const shift = state.tShift[startLine];
+        const max = state.eMarks[startLine];
+        let pos = state.bMarks[startLine];
+        pos += shift;
+
+        if (shift > 3 || pos + 2 >= max) { return false; }
```

Comment:
I think some additional comments would be good here, what do slight, pos and max mean?